Library/Homebrew/extend/pathname.rb

```diff
@@ -360,10 +360,12 @@ def write_exec_script *targets
     mkpath
     targets.each do |target|
       target = Pathname.new(target) # allow pathnames or strings
-      (self+target.basename()).write <<-EOS.undent
-        #!/bin/bash
-        exec "#{target}" "$@"
-      EOS
+      if File.stat(target).file? && File.stat(target).executable?
```

Comment:
It depends on whether or not one prefers ["standard Ruby"](http://ruby-doc.org/stdlib-2.1.0/libdoc/pathname/rdoc/Pathname.html#class-Pathname-label-Example+2%3A+Using+standard+Ruby) or [a façade](http://ruby-doc.org/stdlib-2.1.0/libdoc/pathname/rdoc/Pathname.html#class-Pathname-label-File+status+predicate+methods). :wink:

Kidding aside, I prefer the explicit version since `target.file?` and `target.executable?` seems like it could be confused for a custom method checking for a "target file" and "target executable".